src/app/calendar/getParcelCalendarEvents.ts

```diff
@@ -0,0 +1,87 @@
+import supabase, { Schema } from "@/supabase";
+import { CalendarEvent } from "@/components/Calendar/Calendar";
+
+const COLLECTION_DURATION_MS = 30 * 60 * 1000;
+
+interface LocationColorMap {
+    [location: string]: string;
+}
+
+// TODO VFB-16 Consider adding a rainbow palette for a selection of accessible colours (something like these but they don't need to be matched exactly)
+const colorMap: LocationColorMap = {
+    "Brixton Hill - Methodist Church": "#FA9189",
+    "Clapham - St Stephens Church": "#FCAE7C",
+    "N&B - Emmanuel Church": "#FFD868",
+    "Streatham - Immanuel & St Andrew": "#B3F5BC",
+    "Vauxhall Hope Church": "#66B79B",
+    "Waterloo - Oasis": "#A0C7DB",
+    "Waterloo - St George the Martyr": "#96B9FF",
+    "Waterloo - St Johns": "#CEAAE9",
+    Delivery: "#A7A1FF",
+    default: "#D2C9BC",
+};
+
+// TODO VFB-16 Add a font colour that is accessible with the colours within the map above
+const eventTextColor = "#000000";
+
+interface ClientMap {
+    [primary_key: string]: Schema["clients"];
+}
+
+const getClientMap = async (): Promise<ClientMap> => {
+    const response = await supabase.from("clients").select();
+    const clients = response.data ?? [];
+    const clientMap: ClientMap = {};
+
+    clients.forEach((client) => {
+        clientMap[client.primary_key] = client;
+    });
+
+    return clientMap;
+};
+
+const getParcelsWithCollectionDate = async (): Promise<Schema["parcels"][]> => {
+    const response = await supabase.from("parcels").select().not("collection_datetime", "is", null);
+    return response.data ?? [];
+};
+
+const parcelsToCollectionEvents = (
+    parcels: Schema["parcels"][],
+    clientMap: ClientMap
+): CalendarEvent[] => {
+    if (Object.keys(clientMap).length === 0) {
```

Comment:
this condition _shouldn't_ ever happen with our database - happy to keep it (because it gets rid of some errors!) but interested where it came from? Faintly remember introducing it a bit of time ago :)